models/transaction.go

```diff
@@ -86,18 +93,33 @@ func DecodeTransaction(event cadence.Event) (*gethTypes.Transaction, error) {
 	var t txEventPayload
 	err := cadence.DecodeFields(event, &t)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to cadence decode transaction: %w", err)
 	}
 
 	encTx, err := hex.DecodeString(t.Transaction)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to decode transaction hex: %w", err)
+	}
+
+	// check if the transaction data is actually from a direct call, which is a special flow/evm state transition
+	if encTx[0] == types.DirectCallTxType {
+		// todo(sideninja) support indexing of direct calls in the future
```

Comment:
For now it is ok, because it was a blocker by direct calls not having correct IDs. This will be a follow up PR that will index those as well. But no, balances will be fine because they are requested on the network using scripts right now. But good callout for sure.